source/mongodb-handshake/handshake.rst

```diff
@@ -224,6 +224,49 @@ Example::
         - "Oracle JVM EE 9.1.1"
 
 
+Supporting Wrapping Libraries
+=============================
+
+Drivers MUST allow libraries which wrap the driver to append to the client
+metadata generated by the driver. The following class definition defines the
+options which MUST be supported:
+
+.. code:: typescript
+
+  class DriverInfoOptions {
+    /**
+     * The name of the library wrapping the driver.
+     */
+    name: Optional<String>;
+
+    /**
+     * The version of the library wrapping the driver.
+     */
+    version: Optional<String>;
+
+    /**
+     * Optional platform information for the wrapping driver.
+     */
+    platform: Optional<String>;
+  }
+
+Note that how these options are provided to a driver is left up to the implementor.
+
+If provided, these options MUST not replace the values used for metadata generation, but
+instead append to the respective fields, delimited by a `|` character. For example, when
+`Motor <https://docs.mongodb.com/ecosystem/drivers/motor/>`_ wraps PyMongo, the following
+fields are updated to include Motors "driver info":
+
+    {
+        client: {
+            driver: {
+                name: "nodejs|motor",
+                version: "2.6.1|2.0.0"
+            }
+        }
+    }
+
+
 Limitations
```

Comment:
Should drivers perform some validation on the DriverInfo fields to avoid hitting these size limits?

Reply:
I'm not sure we can, other than providing guidance around what fields to prioritize? Would we want to put size limits on actual fields?

Reply:
I'm not for this idea anymore. It would be complicated to compute the actual size limit based on the 512 byte restriction.